src/track/globaltrackcache.cpp

```diff
@@ -420,8 +420,34 @@ bool GlobalTrackCache::isEmpty() const {
     return m_tracksById.empty() && m_tracksByCanonicalLocation.empty();
 }
 
+TrackPointer GlobalTrackCache::lookupByRef(
+        const TrackRef& trackRef) {
+    TrackPointer trackPtr;
+    if (trackRef.hasId()) {
+        trackPtr = lookupById(trackRef.getId());
+        if (trackPtr) {
+            return trackPtr;
+        }
+    }
+    if (trackRef.hasCanonicalLocation()) {
+        trackPtr = lookupByCanonicalLocation(trackRef.getCanonicalLocation());
+        if (trackPtr) {
+            if (trackRef.hasId() &&
```

Comment:
Can this ever be true because of the check in line 426?

Reply:
If another track with a different id but the same canonical location has already been stored in the cache we will get here. That's part of the problem.

Reply:
But we might check that the found track also already has a valid id. Otherwise, this is a rare, intermediate edge case.